src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DataSetDataTableInSerializableTypeTests.cs

```diff
@@ -376,16 +376,127 @@ public class BlahClass
 }");
         }
 
+        [Fact]
+        public async Task GeneratedCode_Diagnostic()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+using System.Data;
+using System.Xml.Serialization;
+
+namespace Blah
+{
+    [global::System.CodeDom.Compiler.GeneratedCode(""System.Data.Design.TypedDataSetGenerator"", ""2.0.0.0"")]
+    [global::System.Serializable()]
+    [global::System.ComponentModel.DesignerCategoryAttribute(""code"")]
+    [global::System.ComponentModel.ToolboxItem(true)]
+    [global::System.Xml.Serialization.XmlSchemaProviderAttribute(""GetTypedDataSetSchema"")]
+    [global::System.Xml.Serialization.XmlRootAttribute(""Package"")]
+    [global::System.ComponentModel.Design.HelpKeywordAttribute(""vs.data.DataSet"")]
+    public class BlahClass : global::System.Data.DataSet {
+        private DataTable table;
+    }
+}",
+                GetAutogeneratedIFormatterCSharpResultAt(7, 5, "DataSet", "BlahClass"),
```

Comment:
[minor][suggestion] It'd be nice to use the markup syntax instead, it makes it easier to see location of the issues and make refactoring easy too.

Reply:
Hrm, I still need to learn me the markup syntax. Will probably address that in a separate PR.